controllers/StockApiController.php

```diff
@@ -282,6 +282,10 @@ public function ConsumeProduct(\Psr\Http\Message\ServerRequestInterface $request
 			{
 				$specificStockEntryId = $requestBody['stock_entry_id'];
 			}
+			elseif (array_key_exists('stock_id', $args) && !empty($args['stock_id']))
+			{
+				$specificStockEntryId = $args['stock_id'];
+			}
```

Comment:
I don't think we need this 2-step handling (simple is better), if a grocycode was supplied, its stock entry id (if present) should be used and take precedence.

Reply:
Reordered the precedence in the new commit - grocycode stock_id's now go first